notification-channel-config-service-impl/src/main/java/org/hypertrace/notification/config/service/Utils.java

```diff
@@ -0,0 +1,107 @@
+package org.hypertrace.notification.config.service;
+
+import io.grpc.Status;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.SneakyThrows;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.notification.config.service.v1.NotificationChannel;
+import org.hypertrace.notification.config.service.v1.NotificationChannelMutableData;
+import org.hypertrace.notification.config.service.v1.WebhookChannelConfig;
+import org.hypertrace.notification.config.service.v1.WebhookHeader;
+
+public class Utils {
+
+  private final NotificationChannelStore notificationChannelStore;
+
+  public Utils(NotificationChannelStore notificationChannelStore) {
```

Comment:
The fact that this takes in a store class is a major red flag. Utils are typically pure functions - data in, data out. No state or state changes.

Perhaps this indicates a better place for this logic would be in the store itself? For things like defaulting newly added fields, I would say that's for sure the case. Will read on to see what else this is doing.